src/dataflow/src/source/postgres.rs

```diff
@@ -157,15 +164,39 @@ impl PostgresSourceReader {
                         let a: Datum = row.get(n).into();
                         a
                     }));
-                    snapshot_tx.insert(mz_row).await?;
+                    snapshot_tx.insert(mz_row.clone()).await?;
+                    writeln!(buffer, "{}", serde_json::to_string(&mz_row)?)?;
```

Comment:
the file looks like this for a 4-byte INTEGER and a 7-byte TEXT column:


```
...
{"data":[3,5,64,0,0,23,12,0,0,0,0,0,0,0,17,1,57,17,7,97,98,99,10,120,121,122]}
{"data":[3,5,64,0,0,23,13,0,0,0,0,0,0,0,17,2,49,48,17,7,97,98,99,10,120,121,122]}
...
{code}

That seems to be quite verbose. disk usage will be a multiple of the size of the original data as stored in Postgres and it will take a while to process.

Reply:
Maybe we could go with [`bincode`](https://github.com/bincode-org/bincode)? Looks pretty compact and has a drop-in API

Reply:
We would need to change the serialization/deserialization of `Row` in order to switch to `bincode`. I am open to it, but I think I'll ping the #engineering channel to check if there's a reason why we wouldn't want to do that!

Reply:
hm, but `Row` implements the Serializable trait which I assume means we can serialize with anything, no?

Reply:
ack - you're right! fixed

Reply:
For future reference, testing this out with a tiny amount of data:
```
materialize=> select * from t1;
  oid  | row_data
-------+-----------
 34121 | {hi,1}
 34121 | {hello,2}
(2 rows)
```
`serde` writes 139 bytes and `bincode` writes 61! thanks for the suggestion @philip-stoev and @petrosagg!